File: Week_1/Java_Programming/Basic_Queries_Array.java
import java.util.Scanner;

public class Basic_Queries_Array {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int max = -1000;
        int min = 1000;
        int sum = 0;
        int[] array = new int[n];
        for(int i=0; i<n; i++){
            array[i] = sc.nextInt();
            sum += array[i];
            if(array[i] > max) max = array[i];
            if(array[i] < min) min = array[i];
        }
        String endArray = sc.nextLine();
        while(sc.hasNextLine()){
            int max1 = -1000;
            String Command = sc.nextLine();
            String[] parts = Command.split(" ");
            if(parts[0].equals("find-max")) System.out.println(max);
            else if(parts[0].equals("find-min")) System.out.println(min);
            else if(parts[0].equals("sum")) System.out.println(sum);
            else if(parts[0].equals("find-max-segment")){
                int a = Integer.parseInt(parts[1]);
                int b = Integer.parseInt(parts[2]);
                for(int i=a-1; i<b; i++){
                    if(array[i]>max1) max1 = array[i];
                }
                System.out.println(max1);
            }
            else if(parts[0].equals("*")) continue;
            else break;
        }
    }
}

File: Week_1/C_Programming/Basic_Queries_Array.cpp
#include <stdio.h>
#include <string.h>

int main() {
    int n;
    int sum = 0; 
    int max = -1000;
    int min = 1000;
    scanf("%d", &n);
    int a[n];
    for (int i = 1; i <= n; i++){
    	scanf("%d", &a[i]);
    	if(max<a[i]) max = a[i];
    	if(min>a[i]) min =a[i];
    	sum = sum+a[i];
	}
    char s[15];
    while (1) {
        scanf("%s", s);
        if (strcmp(s, "find-max") == 0)
            printf("%d\n", max);
        else if (strcmp(s, "find-min") == 0) 
            printf("%d\n", min);
        else if (strcmp(s, "find-max-segment") == 0) {
            int x, y;
            int ans = -1000;
            scanf("%d %d", &x, &y);
            for (int i = x; i <= y; i++)
                if(ans<a[i]) ans=a[i];
            printf("%d\n", ans);
        }
        else if (strcmp(s, "sum") == 0) 
            printf("%d\n", sum);
        else if(strcmp(s, "***") == 0) break;
        else continue;
    }
    return 0;
}
